pkg/operator/observe_config.go

```diff
@@ -92,12 +84,42 @@ func (c ConfigObserver) sync() error {
 	// don't worry about errors
 	currentConfig := map[string]interface{}{}
 	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
+	observedConfig := map[string]interface{}{}
+	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&observedConfig)
+
+	var errors []error
+	for _, observer := range c.observers {
+		var e []error
+		observedConfig, e = observer(c.kubeClient, &rest.Config{}, observedConfig)
+		errors = append(errors, e...)
+	}
 	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+		glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, observedConfig))
+		operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: observedConfig}}
+	}
+	if len(errors) > 0 {
+		var message bytes.Buffer
+		var i int
+		for i, err = range errors {
+			if i > 0 {
+				message.WriteString("\n")
```

Comment:
readable above all else until you prove this is bad in a profile..  make a slice of strings from the errors, then `strings.Join`